sycl/include/CL/sycl/detail/device_info.hpp

```diff
@@ -289,6 +290,34 @@ struct get_device_info<vector_class<size_t>,
   }
 };
 
+// Specialization for kernel to kernel pipes.
+// Here we step away from OpenCL, since there is no appropriate cl_device_info
+// enum for global pipes feature.
+template <>
+struct get_device_info<bool, info::device::kernel_kernel_pipe_support> {
+  static bool _(RT::PiDevice dev) {
+    // If a device's backend supports SPIR-V 1.1 (where Pipe Storage feature was
+    // defined), than it supports the feature as well.
+    string_class spirv_version =
+      get_device_info<string_class, info::device::spirv_version>()._(dev);
+    if (spirv_version.find("SPIR-V_1.1") != string_class::npos)
+      return true;
+
+    // Also we claim, that all Intel FPGA devices support kernel to kernel pipe
+    // feature (at least at the scope of SYCL_INTEL_data_flow_pipes extension).
+    info::device_type device_type =
+      get_device_info<info::device_type, info::device::device_type>()._(dev);
+    string_class vendor_name =
+      get_device_info<string_class, info::device::vendor>()._(dev);
+    if (device_type == info::device_type::accelerator &&
```

Comment:
There should be a device selector for IntelFPGA, which has a more precise logic. Can you re-use it here?

Reply:
Done